backend/presentation/tests.py

```diff
@@ -0,0 +1,325 @@
+            path=self.detail_url, data=self.updated_presentation_data, format="json"
+        )
+        self.presentation_1.refresh_from_db()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.data, PresentationSerializer(self.presentation_1).data
+        )
+
+    def test_patch_presentation_detail_without_authentication(self):
+        response = self.client.patch(
+            path=self.detail_url, data=self.updated_presentation_data, format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_patch_presentation_detail_as_not_owner(self):
+        self.client.force_authenticate(user=self.user_2)
+        response = self.client.patch(
+            path=self.detail_url, data=self.updated_presentation_data, format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_patch_presentation_detail_as_owner(self):
+        self.client.force_authenticate(user=self.user_1)
+        response = self.client.patch(
+            path=self.detail_url, data=self.updated_presentation_data, format="json"
+        )
+        self.presentation_1.refresh_from_db()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.data, PresentationSerializer(self.presentation_1).data
+        )
+
+    def test_patch_presentation_detail_as_admin(self):
+        self.client.force_authenticate(user=self.admin)
+        response = self.client.patch(
+            path=self.detail_url, data=self.updated_presentation_data, format="json"
+        )
+        self.presentation_1.refresh_from_db()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.data, PresentationSerializer(self.presentation_1).data
+        )
+
+    def test_delete_presentation_without_auth(self):
+        response = self.client.delete(path=self.detail_url)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_delete_presentation_as_not_owner(self):
+        self.client.force_authenticate(user=self.user_2)
+        response = self.client.delete(path=self.detail_url)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_delete_presentation_as_owner(self):
+        self.client.force_authenticate(user=self.user_1)
+        response = self.client.delete(path=self.detail_url)
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(Presentation.objects.all().count(), 0)
+
+    def test_delete_presentation_as_admin(self):
+        self.client.force_authenticate(user=self.admin)
+        response = self.client.delete(path=self.detail_url)
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(Presentation.objects.all().count(), 0)
```

Comment:
```suggestion
        self.assertEqual(Presentation.objects.count(), 0)
```
Previous version will not only get the count, but also fetch all presentation model fields.